go-controller/pkg/util/pod_annotation.go

```diff
@@ -120,13 +126,18 @@ func MarshalPodAnnotation(annotations map[string]string, podInfo *PodAnnotation,
 
 	existingPa, ok := podNetworks[nadName]
 	if ok {
-		if len(pa.IPs) != len(existingPa.IPs) {
-			return nil, ErrOverridePodIPs
-		}
-		for _, ip := range pa.IPs {
-			if !SliceHasStringItem(existingPa.IPs, ip) {
+		if len(existingPa.IPs) > 0 {
```

Comment:
why is this needed ? I would expect len(pa.IPs) to be == to len(existingPa.IPs) == 0.